hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/DefinitionsService.java

```diff
@@ -38,39 +39,108 @@
+     * defined a call to {@link #updateGroupTrigger(String, Trigger)} is needed to enable the <code>Trigger</code>.
+     * <p>
+     * A non-group trigger can never be made into a group trigger, and vice-versa.
+     * </p>
+     * @param tenantId Tenant where trigger is created
+     * @param groupTrigger New trigger definition to be added
+     * @throws Exception If the <code>Trigger</code> already exists.
+     * @see {@link #addTrigger(String, Trigger)} for adding a non-group trigger.
+     */
+    void addGroupTrigger(String tenantId, Trigger groupTrigger) throws Exception;
+
+    /**
+     * Generate a member trigger for the specified group trigger. The dataIdMap replaces the tokens in the
+     * group trigger's conditions with actual dataIds. The member trigger gets the enabled state of the group.
+     * @param tenantId Tenant where trigger is stored
+     * @param groupId Group triggerId from which to spawn the member trigger
+     * @param memberId The member triggerId, unique id within the tenant, if null an Id will be generated
+     * @param memberName The member triggerName, not null, unique name within the tenant
+     * @param memberContext The member triggerContext. If null the context is inherited from the group trigger
+     * @param dataIdMap Tokens to be replaced in the new trigger
+     * @return the member trigger
+     * @throws Exception on any problem
+     * @see {@link #addTrigger(String, Trigger)} for adding a non-group trigger.
+     * @see {@link #addGroupTrigger(String, Trigger)} for adding a group trigger.
+     */
+    Trigger addMemberTrigger(String tenantId, String groupId, String memberId, String memberName,
+            Map<String, String> memberContext, Map<String, String> dataIdMap) throws Exception;
+
+    /**
      * The <code>Trigger</code> will be removed from the Alerts engine, as needed, and will no longer be persisted.
      * @param tenantId Tenant where trigger is stored
      * @param triggerId Trigger to be removed
+     * @throws NotFoundException if trigger is not found
      * @throws Exception on any problem
+     * @see {@link #removeGroupTrigger(String, String, boolean, boolean)} for removing a group trigger and its members.
      */
     void removeTrigger(String tenantId, String triggerId) throws Exception;
 
     /**
-     * Update the <code>Trigger</code>. <code>Conditions</code> and <code>Actions</code> are manipulated in separate
-     * calls. The updated <code>Trigger</code> will be persisted.  If enabled the <code>Trigger</code>
-     * will be [re-]inserted into the Alerts engine and any prior dampening will be reset.
+     * The group <code>Trigger</code> will be removed from the Alerts engine, as needed, and will no longer be
+     * persisted. The member triggers will be removed as well, depending on the settings for
+     * <code>leaveMember</code> and <code>leaveOrphans</code>. Note that any member triggers not removed will
```

Comment:
Javadoc needs to be updated. It maintains leave* for parameters in the signature.